plugin/src/main/kotlin/com/jraska/module/graph/plugin/GradleDependencyGraphFactory.kt

```diff
@@ -0,0 +1,33 @@
+package com.jraska.module.graph.plugin
+
+import com.jraska.module.graph.DependencyGraph
+import org.gradle.api.Project
+import org.gradle.api.internal.artifacts.dependencies.DefaultProjectDependency
+
+object GradleDependencyGraphFactory {
+
+  fun create(project: Project): DependencyGraph {
+    val dependencies = project.listDependencyPairs()
+    return DependencyGraph.create(dependencies)
+  }
+
+  private fun Project.listDependencyPairs(): List<Pair<String, String>> {
+    val configurationToLook = setOf("implementation", "api")
+    return rootProject.subprojects
+      .flatMap { project ->
+        project.configurations
+          .filter { configurationToLook.contains(it.name) }
+          .flatMap { configuration ->
+            configuration.dependencies.filterIsInstance(DefaultProjectDependency::class.java)
```

Comment:
It's safer to check the interface `ProjectDependency` here.
`DefaultProjectDependency` is only one implementation.
E.g. for type-safe project accessors there will be another implementation.